tests/datavolume_test.go

```diff
@@ -1119,6 +1121,90 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
+			err := client.Get(context.TODO(), types.NamespacedName{Name: name}, storageProfile)
+			Expect(err).ToNot(HaveOccurred())
+			storageProfile.Spec = spec
+			err = client.Update(context.TODO(), storageProfile)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		It("succeeds creating a PVC from DV without accessModes", func() {
+			defaultScName := utils.DefaultStorageClass.GetName()
+
+			By(fmt.Sprintf("configure storage profile %s", defaultScName))
+			originalProfileSpec := getStorageProfileSpec(f.CrClient, defaultScName)
+			accessModes := []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}
+			volumeMode := v1.PersistentVolumeFilesystem
+			propertySet := cdiv1.ClaimPropertySet{AccessModes: accessModes, VolumeMode: &volumeMode}
+			updateStorageProfileSpec(f.CrClient,
+				defaultScName,
+				cdiv1.StorageProfileSpec{ClaimPropertySets: []cdiv1.ClaimPropertySet{propertySet}})
+
+			By("Verify propertySet updated")
+			Eventually(func() cdiv1.ClaimPropertySet {
+				profile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), defaultScName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				if len(profile.Status.ClaimPropertySets) > 0 {
+					return profile.Status.ClaimPropertySets[0]
+				}
+				return cdiv1.ClaimPropertySet{}
+			}, time.Second*30, time.Second).Should(Equal(propertySet))
+
+			By(fmt.Sprintf("creating new datavolume %s without accessModes", dataVolumeName))
+			dataVolume := createDataVolume(f, defaultScName)
+
+			By("verifying pvc created with correct accessModes")
+			pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pvc.Spec.AccessModes).To(Equal(accessModes))
+
+			By("Restore the profile")
+			updateStorageProfileSpec(f.CrClient, defaultScName, *originalProfileSpec)
+		})
+
+		It("fails creating a PVC from DV without accessModes, no profile", func() {
+			// assumes local is available and has no volumeMode
+			defaultScName := "local"
+			By(fmt.Sprintf("creating new datavolume %s without accessModes", dataVolumeName))
+			dataVolume := createDataVolume(f, defaultScName)
+
+			By("verifying pvc created")
+			_, err := utils.FindPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+
+			By(fmt.Sprint("verifying event occurred"))
+			Eventually(func() bool {
+				// Only find DV events, we know the PVC gets the same events
+				events, err := RunKubectlCommand(f, "get", "events", "-n", dataVolume.Namespace, "--field-selector=involvedObject.kind=DataVolume")
+				if err == nil {
+					fmt.Fprintf(GinkgoWriter, "%s", events)
+					return strings.Contains(events, controller.ErrClaimNotValid) && strings.Contains(events, "DataVolume spec is missing accessMode and cannot get access mode from StorageProfile")
+				}
+				fmt.Fprintf(GinkgoWriter, "ERROR: %s\n", err.Error())
+				return false
+			}, timeout, pollingInterval).Should(BeTrue())
+		})
```

Comment:
Can we have a test that checks that the DV recovers if the user edits the StorageProfile to include the recommended modes in the spec?

Reply:
Done